test/e2e/publish_confirm.test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect } from "chai"
+import { randomUUID } from "crypto"
+import { Connection, ListenersParams } from "../../src"
+import { Rabbit } from "../support/rabbit"
+import { eventually, password, username } from "../support/util"
+import { createConnection } from "../support/fake_data"
+
+describe("publish a message and get confirmation", () => {
+  const rabbit = new Rabbit(username, password)
+  let connection: Connection
+  let confirmed: boolean
+  let stream: string
+  const publisherRef = "my publisher"
+  const listeners: ListenersParams = {
+    metadata_update: (_data) => {
+      return
+    },
+    publish_confirm: (_data) => (confirmed = true),
```

Comment:
I suggest to collect the data in and array and reset before each test to verify in the test